src/renderers/dom/shared/CSSPropertyOperations.js

```diff
@@ -106,6 +120,10 @@ if (__DEV__) {
     } else if (badStyleValueWithSemicolonPattern.test(value)) {
       warnStyleValueWithSemicolon(name, value);
     }
+
+    if (numberIsNaN(value)) {
```

Comment:
Don't inline the short one then. `if (typeof value === 'number' && isNaN(value))`